snakemake/jobs.py

```diff
@@ -551,6 +542,41 @@ def output_mintime(self):
 
         return mintime
 
+    def missing_requested_output(self, requested_output):
```

Comment:
I do not yet understand why it is beneficial to have this quasi duplicated method. Why not just take the original one. Or alternatively, reimplement the original one while specifying job.output as requested_output.

Reply:
The original method was is called in two different scenarios.

In the first, all the outputs are checked. In this case, the original method is a little more efficient.

In the second, only one file or a few files is/are checked at a time. This is what I wrote the new method for.  

If the first case is not a bottleneck, it would be OK to use the new method for both, but I haven't checked that.

Reply:
I have set up a comprehensive benchmark for PR #710. Would it be ok for you if I pick individual changes from this PR and include them manually? Of course, I will give you the credit in the commit messages.

Reply:
And in the release.